### PR TITLE
build(ci): Provide diff for pre-commit failures

### PR DESCRIPTION
**Pull request by jsoref**
### SUMMARY

when pre-commit fails, it's typically because of changes made by one of the tools. As it *knows* what the changes are, this change provides access to people so that they can use `patch -p1` to apply the changes instead of just being frustrated by an unhelpful CI.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

Before:
https://github.com/apache/superset/actions/runs/6507284267/job/17674397883
image

After:
https://github.com/check-spelling-sandbox/superset/actions/runs/6509697746/job/17681762469
image

### TESTING INSTRUCTIONS
1. Make a change to a helm chart that's incomplete such that pre-commit will fail.
2. Make a PR to a branch with this modified workflow 

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by craig-rueda — APPROVED** [review]
LGTM! Let's try this out

**Review by rusackas — APPROVED** [review]
Thanks! Pinging @craig-rueda to see if he has any objections since he knows more about the helm risks than I do.

File: .github/workflows/superset-python-misc.yml [hunk 1]
```diff
@@ -86,7 +86,12 @@ jobs:
           echo "HOMEBREW_REPOSITORY=$HOMEBREW_REPOSITORY" >>"${GITHUB_ENV}"
           brew install norwoodj/tap/helm-docs
       - name: pre-commit
-        run: pre-commit run --all-files
```
Comment (john-bodley) [inline .github/workflows/superset-python-misc.yml:89]: @jsoref would the `pre-commit run` `--show-diff-on-failure` flag be suffice?

Comment (jsoref) [reply .github/workflows/superset-python-misc.yml:89]: Maybe. I don't use this tool. As long as I can apply the diff and get it not to fail going forward.

Comment (craig-rueda) [reply .github/workflows/superset-python-misc.yml:89]: Seems like a good idea. I don't know much about pre commit tbh

**Pull request merged**